SerialPrograms/Source/PokemonSV/Programs/General/PokemonSV_TournamentFarmer.cpp

```diff
@@ -0,0 +1,370 @@
+
+namespace PokemonAutomation {
+namespace NintendoSwitch {
+namespace PokemonSV {
+
+using namespace Pokemon;
+
+TournamentFarmer_Descriptor::TournamentFarmer_Descriptor()
+    : SingleSwitchProgramDescriptor(
+        "PokemonSV:TournamentFarmer",
+        STRING_POKEMON + " SV", "Tournament Farmer",
+        "ComputerControl/blob/master/Wiki/Programs/PokemonSV/TournamentFarmer.md",
+        "Farm the Ace Academy Tournament for money and prizes.",
+        FeedbackType::REQUIRED,
+        AllowCommandsWhenRunning::DISABLE_COMMANDS,
+        PABotBaseLevel::PABOTBASE_12KB
+    )
+{}
+
+struct TournamentFarmer_Descriptor::Stats : public StatsTracker {
+    Stats()
+        : tournaments(m_stats["Tournaments won"])
+        , battles(m_stats["Battles fought"])
+        , losses(m_stats["Losses"])
+        , errors(m_stats["Errors"])
+    {
+        m_display_order.emplace_back("Tournaments won");
+        m_display_order.emplace_back("Battles fought");
+        m_display_order.emplace_back("Losses");
+        m_display_order.emplace_back("Errors", true);
+    }
+    std::atomic<uint64_t>& tournaments;
+    std::atomic<uint64_t>& battles;
+    std::atomic<uint64_t>& losses;
+    std::atomic<uint64_t>& errors;
+};
+std::unique_ptr<StatsTracker> TournamentFarmer_Descriptor::make_stats() const {
+    return std::unique_ptr<StatsTracker>(new Stats());
+}
+
+TournamentFarmer::TournamentFarmer()
+    : NUM_ROUNDS(
+          "<b>Number of Tournaments to run:",
+          LockWhileRunning::UNLOCKED,
+          100, 0
+          )
+    , TRY_TO_TERASTILLIZE(
+          "<b>Use Terastillization:</b><br>Tera at the start of battle. Will take longer but may be worth the attack boost.",
+          LockWhileRunning::UNLOCKED,
+          false
+          )
+    , SAVE_NUM_ROUNDS(
+          "<b>Save every this many tournaments:</b><br>(zero disables saving)",
+          LockWhileRunning::UNLOCKED,
+          0, 0
+          )
+    , GO_HOME_WHEN_DONE(false)
+    , LANGUAGE(
+          "<b>Game Language:</b><br>The language is needed to read the prizes.",
+          TournamentPrizeNameReader::instance().languages(),
+          LockWhileRunning::LOCKED,
+          true
+          )
+    , TARGET_ITEMS("<b>Items:</b>")
+    , NOTIFICATION_PRIZE_MATCH("Matching Prize", true, false, ImageAttachmentMode::JPG, { "Notifs" })
+    , NOTIFICATION_STATUS_UPDATE("Status Update", true, false, std::chrono::seconds(3600))
+    , NOTIFICATIONS({
+        &NOTIFICATION_PRIZE_MATCH,
+        &NOTIFICATION_STATUS_UPDATE,
+        &NOTIFICATION_PROGRAM_FINISH,
+        &NOTIFICATION_ERROR_FATAL,
+        })
+{
+    PA_ADD_OPTION(NUM_ROUNDS);
+    PA_ADD_OPTION(TRY_TO_TERASTILLIZE);
+    PA_ADD_OPTION(SAVE_NUM_ROUNDS);
+    PA_ADD_OPTION(GO_HOME_WHEN_DONE);
+    PA_ADD_OPTION(LANGUAGE);
+    PA_ADD_OPTION(TARGET_ITEMS);
+    PA_ADD_OPTION(NOTIFICATIONS);
+}
+
+void TournamentFarmer::program(SingleSwitchProgramEnvironment& env, BotBaseContext& context) {
```

Comment:
could you break that function into smaller blocks ? Like one block for one fight, one block for battle lost management, etc ? It'd be easier to understand what's going on